src/HeaderSort/components/HeaderSortContainer.ts

```diff
@@ -0,0 +1,157 @@
+
+import { HeaderSort, HeaderSortProps, SortOrder } from "./HeaderSort";
+
+import "../ui/HeaderSort.scss";
+
+export interface ContainerProps extends WrapperProps {
+    entity: string;
+    caption: string;
+    initialSorted: boolean;
+    sortAttribute: string;
+    sortOrder: "asc" | "desc";
+}
+
+export interface ContainerState {
+    alertMessage?: string;
+    listViewAvailable: boolean;
+    publishedSortAttribute?: string;
+    publishedSortOrder?: SortOrder;
+    publishedSortWidgetFriendlyId?: string;
+    targetListView?: ListView | null;
+    targetNode?: HTMLElement;
+}
+
+export default class HeaderSortContainer extends Component<ContainerProps, ContainerState> {
+    private navigationHandler: object;
+    private dataSourceHelper: DataSourceHelper;
+
+    constructor(props: ContainerProps) {
+        super(props);
+
+        this.state = { listViewAvailable: false };
+
+        this.updateSort = this.updateSort.bind(this);
+        this.navigationHandler = dojoConnect.connect(props.mxform, "onNavigation", this, this.connectToListView.bind(this));
+        this.subScribeToWidgetChanges = this.subScribeToWidgetChanges.bind(this);
+        this.publishWidgetChanges = this.publishWidgetChanges.bind(this);
+    }
+
+    render() {
+        return createElement("div", {
+                className: classNames("widget-header-sort", this.props.class),
+                style: SharedUtils.parseStyle(this.props.style)
+            },
+            createElement(Alert, {
+                bootstrapStyle: "danger",
+                className: "widget-header-sort-alert",
+                message: this.state.alertMessage
+            }),
+            this.renderSort()
+        );
+    }
+
+    componentDidUpdate(_prevProps: ContainerProps, prevState: ContainerState) {
+        if (this.state.listViewAvailable && !prevState.listViewAvailable) {
+            if (this.props.initialSorted) {
+                this.updateSort(this.props.sortAttribute, this.props.sortOrder);
+            }
+        }
+    }
+
+    componentWillUnmount() {
+        dojoConnect.disconnect(this.navigationHandler);
+    }
+
+    private renderSort(): ReactElement<HeaderSortProps> | null {
+        if (!this.state.alertMessage) {
+            return createElement(HeaderSort, {
+                caption: this.props.caption,
+                friendlyId: this.props.friendlyId,
+                initialSorted: this.props.initialSorted,
+                onClickAction: this.updateSort,
+                publishedSortAttribute: this.state.publishedSortAttribute,
+                publishedSortOrder: this.state.publishedSortOrder,
+                publishedSortWidgetFriendlyId: this.state.publishedSortWidgetFriendlyId,
+                sortAttribute: this.props.sortAttribute,
+                sortOrder: this.initialSortOrder(this.props.initialSorted, this.props.sortOrder)
+            });
+        }
+
+        return null;
+    }
+
+    private connectToListView() {
```

Comment:
Can we extract the comment part of this function, which is shared across all list view components and make an static function on the `DataSourceHelper`?

Reply:
Was done. 
It was a major change. needs thorough testing regarding hiding of content before updating of listview